iota-client/src/client.rs

```diff
@@ -108,37 +116,72 @@ impl std::fmt::Debug for Client {
     }
 }
 
+impl Drop for Client {
+    /// Gracefully shutdown the `Client`
+    fn drop(&mut self) {
+        self.stop_sync.store(true, Ordering::Relaxed);
+        if let Some(handle) = self.sync_handle.take() {
+            handle.join().expect("failed to join node sync thread");
+        }
+    }
+}
+
 impl Client {
     /// Create the builder to instntiate the IOTA Client.
     pub fn builder() -> ClientBuilder {
         ClientBuilder::new()
     }
 
-    // TODO Implement syncing process
+    /// Sync the node lists per node_sync_interval milliseconds
+    pub(crate) fn start_sync_process(
+        mut runtime: Runtime,
+        sync: Arc<RwLock<HashSet<Url>>>,
+        nodes: Vec<Url>,
+        node_sync_interval: NonZeroU64,
+        stop_sync: Arc<AtomicBool>,
+    ) -> thread::JoinHandle<()> {
+        let node_sync_interval = Duration::from_millis(node_sync_interval.into());
+
+        thread::spawn(move || {
+            while !stop_sync.load(Ordering::Relaxed) {
+                // sleep first since the first `sync_nodes` call is made by the builder
+                // to ensure the node list is filled before the client is used
+                thread::sleep(node_sync_interval);
+                Client::sync_nodes(&mut runtime, &sync, &nodes);
+            }
+        })
+    }
 
-    // pub(crate) fn sync(&mut self) {
-    //     let mut sync_list: HashMap<usize, Vec<Url>> = HashMap::new();
-    //     for url in &*self.pool.read().unwrap() {
-    //         if let Ok(milestone) = self.get_info(url.clone()) {
-    //             let set = sync_list
-    //                 .entry(milestone.latest_milestone_index)
-    //                 .or_insert(Vec::new());
-    //             set.push(url.clone());
-    //         };
-    //     }
+    pub(crate) fn sync_nodes(
+        runtime: &mut Runtime,
+        sync: &Arc<RwLock<HashSet<Url>>>,
+        nodes: &[Url],
+    ) {
+        let mut synced_nodes = HashSet::new();
+
+        runtime.block_on(async {
```

Comment:
We're sure we want to block on this, right ?

Reply:
yeah it's important specially on the first run